src/navigators/index.js

```diff
@@ -1,29 +1,28 @@
-import { createAppContainer } from 'react-navigation';
-import { createStackNavigator } from 'react-navigation-stack';
-import createAnimatedSwitchNavigator from 'react-navigation-animated-switch';
-import LoginScreen from 'screens/LoginScreen';
-import MainScreen from 'screens/MainScreen';
-import SignUpScreen from 'screens/SignUpScreen';
-import AppLoader from 'screens/AppLoader';
+import React from 'react';
+import { NavigationContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
 
-const AuthStack = createStackNavigator({
-  LoginScreen,
-  SignUpScreen,
-});
+import useSession from 'hooks/useSession';
 
-const AppStack = createStackNavigator({
-  MainScreen,
-});
+import AppStack from './AppStack';
+import AuthStack from './AuthStack';
 
-const AppNavigator = createAnimatedSwitchNavigator(
-  {
-    AppLoader,
-    AuthStack,
-    AppStack,
-  },
-  {
-    initialRouteName: 'AppLoader',
-  },
-);
+const Stack = createStackNavigator();
 
-export default createAppContainer(AppNavigator);
+const Navigation = () => {
+  const { user, info } = useSession();
+
+  return (
+    <NavigationContainer>
+      <Stack.Navigator headerMode="none">
+        {user && info ? (
+          <Stack.Screen name="AppStack" component={AppStack} />
```

Comment:
Just a suggestion, but it's a good idea to have all screen/stacks names on constants, so it's easier to match names and navigations

Reply:
Cool